controllers/snapshot/snapshot_adapter_test.go

```diff
@@ -732,6 +791,90 @@ var _ = Describe("Snapshot Adapter", Ordered, func() {
 			Expect(found).To(BeFalse())
 
 		})
+
+		It("Ensure error is logged when experiencing error when fetching ITS for application", func() {
+			var buf bytes.Buffer
+			log := helpers.IntegrationLogger{Logger: buflogr.NewWithBuffer(&buf)}
+			adapter = NewAdapter(hasSnapshot, hasApp, hasComp, log, loader.NewMockLoader(), k8sClient, ctx)
+			adapter.context = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ApplicationContextKey,
+					Resource:   hasApp,
+				},
+				{
+					ContextKey: loader.ComponentContextKey,
+					Resource:   hasComp,
+				},
+				{
+					ContextKey: loader.SnapshotContextKey,
+					Resource:   hasSnapshot,
+				},
+				{
+					ContextKey: loader.AllIntegrationTestScenariosContextKey,
+					Err:        fmt.Errorf("not found"),
+				},
+				{
+					ContextKey: loader.RequiredIntegrationTestScenariosContextKey,
+					Err:        fmt.Errorf("not found"),
+				},
+			})
+			_, _ = adapter.EnsureStaticIntegrationPipelineRunsExist()
+			Expect(buf.String()).Should(ContainSubstring("Failed to get Integration test scenarios for the following application"))
+			Expect(buf.String()).Should(ContainSubstring("Failed to get all required IntegrationTestScenarios"))
+		})
+
+		It("Mark snapshot as pass when required ITS is not found", func() {
+			var buf bytes.Buffer
+			log := helpers.IntegrationLogger{Logger: buflogr.NewWithBuffer(&buf)}
+			adapter = NewAdapter(hasSnapshot, hasApp, hasComp, log, loader.NewMockLoader(), k8sClient, ctx)
+			adapter.context = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ApplicationContextKey,
+					Resource:   hasApp,
+				},
+				{
+					ContextKey: loader.ComponentContextKey,
+					Resource:   hasComp,
+				},
+				{
+					ContextKey: loader.SnapshotContextKey,
+					Resource:   hasSnapshot,
+				},
+				{
+					ContextKey: loader.RequiredIntegrationTestScenariosContextKey,
+					Resource:   nil,
+				},
+			})
+			_, _ = adapter.EnsureStaticIntegrationPipelineRunsExist()
```

Comment:
Why return values aren't checked?

Reply:
Added. Actually the result and err after the `if..else` has been checked several times.